include/bl_partitions.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef BL_PARTITIONS_H
+#define BL_PARTITIONS_H
+
+#if defined(CONFIG_PARTITION_MANAGER_ENABLED)
+#include <pm_config.h>
+#define NSIB_B0_ADDRESS PM_B0_ADDRESS
+#define NSIB_B0_SIZE PM_B0_SIZE
+#define NSIB_PROVISION_ADDRESS PM_PROVISION_ADDRESS
+#define NSIB_PROVISION_SIZE PM_PROVISION_SIZE
+
+#if defined(PM_S0_ADDRESS)
+#define NSIB_S0_ADDRESS PM_S0_ADDRESS
+#define NSIB_S0_SIZE PM_S0_SIZE
+#endif
+#if defined(PM_S1_ADDRESS)
+#define NSIB_S1_ADDRESS PM_S1_ADDRESS
+#define NSIB_S1_SIZE PM_S1_SIZE
+#endif
+
+#else /* defined(CONFIG_PARTITION_MANAGER_ENABLED) */
+#define NSIB_B0_CONTAINER_NODE B0 DT_PARENT(DT_PARENT(DT_NODELABEL(b0_partition)))
```

Comment:
Please consider using ``DT_GPARENT(..)`` instead of ``DT_PARENT(DT_PARENT(..))``